### PR TITLE
Update mult dir avoid full disk

### PR DESCRIPTION
**Pull request by mdf369**
Currently, Directories doesn't avoid using disks that are full, which is a bug. Therefore, this PR intend to ignore full disks when choosing folder for next tsfile; meanwhile, if disks of all directories are all full, exception will be thrown to warn the users.

### CODE REVIEW

File: iotdb/src/main/java/org/apache/iotdb/db/conf/directories/strategy/MinFolderOccupiedSpaceFirstStrategy.java [hunk 1]
```diff
@@ -22,40 +22,41 @@
...
-        min = current;
-      } else if (min == current) {
-        candidates.add(i);
+  private int getMinOccupiedSpaceFolder() throws DiskSpaceInsufficientException {
+    int minIndex = -1;
+    long minSpace = Integer.MAX_VALUE;
+
+    for (int i = 0; i < folders.size(); i++) {
+      String folder = folders.get(i);
+      if (hasSpace(folder)) {
```
Comment (jixuan1989) [inline iotdb/src/main/java/org/apache/iotdb/db/conf/directories/strategy/MinFolderOccupiedSpaceFirstStrategy.java]: if (!hasSpace(folder)) ?

File: iotdb/src/main/java/org/apache/iotdb/db/conf/directories/strategy/SequenceStrategy.java [hunk 2]
```diff
@@ -19,30 +19,46 @@
...
 
     return index;
   }
 
-  private void updateIndex() {
-    currentIndex++;
-    if (currentIndex >= folders.size()) {
-      currentIndex = 0;
+  private int tryGetNextIndex(int start) throws DiskSpaceInsufficientException {
+    int index = start;
```
Comment (jixuan1989) [inline iotdb/src/main/java/org/apache/iotdb/db/conf/directories/strategy/SequenceStrategy.java]: int index = ( start +1 ) % folders.size(); ?

File: iotdb/src/main/java/org/apache/iotdb/db/conf/directories/strategy/MinFolderOccupiedSpaceFirstStrategy.java (hunk 1 above)
Comment (mdf369) [reply iotdb/src/main/java/org/apache/iotdb/db/conf/directories/strategy/MinFolderOccupiedSpaceFirstStrategy.java]: fixed

File: iotdb/src/main/java/org/apache/iotdb/db/conf/directories/strategy/SequenceStrategy.java (hunk 2 above)
Comment (mdf369) [reply iotdb/src/main/java/org/apache/iotdb/db/conf/directories/strategy/SequenceStrategy.java]: fixed

**Pull request merged**